### DISCUSSION TITLE
How to integrate logto with OpenAPI's OAuth2 flow？

### DISCUSSION THREAD

**Discussion by yixinBC** [logto-io/logto#5459] [Q&A]
I'm trying to use logto in my [FastAPI](https://github.com/tiangolo/fastapi) app.
FastAPI has a feature called ["Automatic docs"](https://fastapi.tiangolo.com/features/#automatic-docs), which is based on OpenAPI
When I followed logto's [Python SDK guide](https://docs.logto.io/sdk/python/#fetch-access-token-for-the-api-resource),I found the SDK will create a client for authentication.I found it's incompatible with FastAPI's OpenAPI OAuth2 flow,which is documented in https://fastapi.tiangolo.com/tutorial/security/.
I tried to integrate logto into a standard OAuth2 flow that can be used for OpenAPI's authentication.I've read the source code of both logto Python SDK and FastAPI's implemetion of OAuth2 in fastapi.security,but just have no idea.
I am far from an Oauth2 or backend-development expert, so if I've made any mistakes,plz just point then out directly.Or is it just impossible to include logto into OpenAPI's OAuth2 flow?

**Comment by simeng-li** [comment]
Hi yixin. Can you give us more details about the FastAPI integration issue you were facing? What specific compatibility issue have you met?
